### PR TITLE
[Explore V2] Events sub-section UI (with mocked data)

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Create the **Events** sub-section in the Explore V2 feature.

**IMPORTANT:** The UI is not yet visually refined, it will be addressed in a separate PR.

### Events sub-section
image

#### UI components to implement in the common library
- [x] Carousel

#### UI components to implement specifically for ExploreV2
- [x] Button_ShowMore
- [x] Button_SubscribeEvent
- [x] Button_UnsubscribeEvent
- [x] CloseEventCardModalButton
- [x] EventCard
- [x] EventCard_Long
- [x] EventCard_Modal
- [x] EventsSubSection

## Related links
- **PRD "Explore V2"**: https://www.notion.so/decentraland/PRD-Explore-V2-ab98e922c196488fb217596c888ce3d6
- **TDD "Explore V2"**: https://www.notion.so/decentraland/TDD-Explore-V2-f2cb9c7ae4404c6f893f537e28423061
- **Figma "Explore V2"**: https://www.figma.com/file/5YhRkm1NXme8MhgDxGT2zY/Explore-V2?node-id=1290%3A4164

## Test Link
https://play.decentraland.zone/?renderer-branch=feat/ExploreV2-Events-Sub-Section-UI
## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentView.cs [hunk 1]
```diff
@@ -13,42 +18,34 @@ public interface IExploreV2MenuComponentView
     /// </summary>
     GameObject go { get; }
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentView.cs:19]: Exposing the GO can lead to bad practices and I'm not sure if it should part of the interface. Do we really need the GameObject to be part of the abstraction?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/PlacesAndEventsSectionComponentController.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+public interface IPlacesAndEventsSectionComponentController
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/PlacesAndEventsSectionComponentController.cs:1]: You can use `IDisposable` here: `IPlacesAndEventsSectionComponentController : IDisposable`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
+using DCL;
+using System.Collections.Generic;
+using UnityEngine;
+
+public interface IEventsSubSectionComponentController
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs:5]: Use `IDisposable` here.

File: unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs [hunk 4]
```diff
@@ -166,18 +198,31 @@ internal IEnumerator InstantiateItemOnEditor(BaseComponentView newItem, string n
...
-                Destroy(child.gameObject);
+                if (Application.isPlaying)
+                {
+                    Destroy(child.gameObject);
+                }
+                else
+                {
+                    if (isActiveAndEnabled)
+                        StartCoroutine(DestroyGameObjectOnEditor(child.gameObject));
+                }
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs:215]: This is quite dangerous. 

If we want to diverge the code from builds to editor let's use preprocessing directives:
```
#if UNITY_EDITOR
  Destroy(child.gameObject);
#else
  StartCoroutine(DestroyGameObjectOnEditor(child.gameObject));
#endif
```

File: unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs [hunk 5]
```diff
@@ -137,11 +157,11 @@ public BaseComponentView GetItem(int index)
 
     public List<BaseComponentView> GetAllItems() { return instantiatedItems; }
 
-    internal void CreateItem(BaseComponentView newItem, string name)
+    internal void CreateItem(BaseComponentView newItem, string name, bool instantiateNewCopyOfItem = true)
     {
         if (Application.isPlaying)
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs:162]: Let's use preprocessor directives for this

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentView.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentView.cs]: You're right, it is not a good practice. Fixed in https://github.com/decentraland/unity-renderer/pull/1238/commits/f83c06799e4871ba364788203e838c0ea0872270. Thanks!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/PlacesAndEventsSectionComponentController.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/PlacesAndEventsSectionComponentController.cs]: Correct! Done in https://github.com/decentraland/unity-renderer/pull/1238/commits/758ac9b508e15b6c9518db4a59c59f4338c26b21. Thanks!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs (hunk 3 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs]: Correct! Done in https://github.com/decentraland/unity-renderer/pull/1238/commits/758ac9b508e15b6c9518db4a59c59f4338c26b21. Thanks!

File: unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs (hunk 4 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs:215]: Here we don't want to separate the build behaviour from the editor one but the inspector behaviour (for the cases when we are configuring an UI component from its inspector and it needs to be refreshed after a frame) from the play mode (where the component refreshing happens normally).

File: unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs (hunk 5 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs:162]: Same as above.

File: unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs (hunk 4 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs:215]: Aaahhh, I would still wrap the part that is only relevant for the Editor in a preprocessing directive. `DestroyImmediate` leads to serious issues and better be safe than sorry and ensuring there's no way we can reach that code in production.

**Review by AjimenezDCL — APPROVED** [review]

**Pull request merged**